### PR TITLE
Upgrade dependencies Flutter and dart

### PR DESCRIPTION
**Pull request by 2002Bishwajeet**
## What does this PR do?

Upgrade dependencies of Flutter and Dart to make it Dart 3.0 Ready

## Test Plan

Manual QA

## Related PRs and Issues

Fixes [#5583](https://github.com/appwrite/appwrite/issues/5583)

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

Yes

### CODE REVIEW

File: templates/dart/pubspec.yaml.twig [hunk 1]
```diff
@@ -8,7 +8,7 @@ documentation: {{ spec.contactURL }}
 environment:
   sdk: '>=2.17.0 <3.0.0'
 dependencies:
-  http: ^0.13.5
+  http: '>=0.13.5 <1.0.1'
```
Comment (lohanidamodar) [inline templates/dart/pubspec.yaml.twig:11]:
```suggestion
  http: '>=0.13.6 <1.0.1'
```

File: templates/flutter/pubspec.yaml.twig [hunk 2]
```diff
@@ -18,17 +18,17 @@ environment:
...
   flutter:
     sdk: flutter
-  cookie_jar: ^4.0.1
-  device_info_plus: ^8.2.0
+  cookie_jar: ^4.0.2
+  device_info_plus: ^9.0.2
   flutter_web_auth_2: ^2.1.4
-  http: ^0.13.5
-  package_info_plus: ^3.1.0
+  http: '>=0.13.5 <1.0.1'
```
Comment (lohanidamodar) [inline templates/flutter/pubspec.yaml.twig:24]:
```suggestion
  http: '>=0.13.6 <1.0.1'
```

**Comment by lohanidamodar** [comment]
@2002Bishwajeet looks good, couple of suggestions and make sure tests are passing.

**Comment by 2002Bishwajeet** [comment]
@lohanidamodar, The tests that failed doesn't seem related to my changes?

File: templates/flutter/pubspec.yaml.twig [hunk 3]
```diff
@@ -18,17 +18,17 @@ environment:
...
-  cookie_jar: ^4.0.1
-  device_info_plus: ^8.2.0
+  cookie_jar: ^4.0.2
+  device_info_plus: ^9.0.2
   flutter_web_auth_2: ^2.1.4
-  http: ^0.13.5
-  package_info_plus: ^3.1.0
+  http: '>=0.13.6 <1.0.1'
+  package_info_plus: ^4.0.2
   path_provider: ^2.0.14
```
Comment (lohanidamodar) [inline templates/flutter/pubspec.yaml.twig:26]:
```suggestion
  path_provider: ^2.0.15
```

File: templates/flutter/pubspec.yaml.twig [hunk 4]
```diff
@@ -18,17 +18,17 @@ environment:
...
   web_socket_channel: ^2.4.0
-  universal_html: ^2.2.1
+  universal_html: ^2.2.2
```
Comment (lohanidamodar) [inline templates/flutter/pubspec.yaml.twig:28]:
```suggestion
  universal_html: ^2.2.3
```

File: templates/flutter/pubspec.yaml.twig [hunk 5]
```diff
@@ -18,17 +18,17 @@ environment:
 dependencies:
   flutter:
     sdk: flutter
-  cookie_jar: ^4.0.1
-  device_info_plus: ^8.2.0
+  cookie_jar: ^4.0.2
```
Comment (lohanidamodar) [inline templates/flutter/pubspec.yaml.twig:21]:
```suggestion
  cookie_jar: ^4.0.3
```

**Review by lohanidamodar — CHANGES_REQUESTED** [review]
some packages have updated since.

**Review by lohanidamodar — APPROVED** [review]

**Pull request merged**